File: tests/fixtures/anti-slop/acceptance/go/round_trip.go
package acceptance

import "reflect"

type Item struct {
	Value int
}

func InterfaceRoundTrip(value Item) Item {
	return any(value).(Item)
}

func ReflectInterfaceRoundTrip(value Item) Item {
	return reflect.ValueOf(value).Interface().(Item)
}

func DirectValue(value Item) Item {
	return value
}

File: tests/fixtures/anti-slop/acceptance/cpp/round_trip.cpp
struct Item {
    int value;
};

Item *redundant_cast(Item *value) {
    return static_cast<Item *>(reinterpret_cast<Item *>(value));
}

Item *void_pointer_chain(Item *value) {
    return static_cast<Item *>(static_cast<void *>(value));
}

Item *direct_value(Item *value) {
    return value;
}
